packages/sitecore-jss-react/src/components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { Component, ErrorInfo, ReactNode } from 'react';
+
+interface ErrorState {
+  error?: Error;
+  errorInfo?: ErrorInfo;
+}
+
+type ErrorComponentProps = {
+  [prop: string]: unknown;
+};
+
+export interface ErrorBoundaryProps {
+  errorComponent?: React.ComponentClass<ErrorComponentProps> | React.FC<ErrorComponentProps>;
+  children?: ReactNode;
+}
+
+//* *
+// * This component will simply wrap any other component passed down to it and contain an error that could be thrown
+// * So that all other components on page would still be displayed
+//*
+export class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorState> {
+  previousError: Error;
+
+  constructor(props: ErrorBoundaryProps) {
+    super(props);
+    this.state = {};
+  }
+
+  defaultErrorComponent: React.FC<{ error: Error; errorInfo: ErrorInfo }> = ({
+    error,
+    errorInfo,
+  }) => {
+    return (
+      <div className="sc-jss-component-error">
+        <p>A rendering error occurred: {error.message}.</p>
+        <p>Stack trace: {errorInfo.componentStack}</p>
+      </div>
+    );
+  };
+
+  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
+    if (JSON.stringify(error) !== JSON.stringify(this.previousError)) {
```

Comment:
What's the reason for tracking/checking previous error? Is it required or extraneous optimization?

Reply:
I noticed that a single error from a single component is reported several times in browser console. As we're updating state when error occurs I thought it's best to only do that once for duplicating errors from the same component.